### PR TITLE
Add Travis CI with Docker

### PR DESCRIPTION
**Pull request by jvaltin**
This builds and outputs a sha256sum of the built `js/backup.js`.

### CODE REVIEW

File: Dockerfile.travis [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+RUN apt update && apt install -y curl gnupg2 git coreutils
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
+RUN apt install -y nodejs
+RUN apt install -y npm
+RUN npm install -g browserify
+
+RUN npm list
+RUN browserify js/backup_in.js -o js/backup.js
+RUN sha256sum js/backup.js
```
Comment (stephaniestroka) [inline Dockerfile.travis:31]: Can you explain to me why you produce the sha256sum of js/backup.js?

**Comment by jvaltin** [comment]
> Can you explain to me why you produce the sha256sum of js/backup.js?

Sure. It is important to produce deterministic outputs for buildable
things. To make that useful, we probably want users to be able to verify
that they are running software that corresponds to source code for any
given commit.
-- 

Thank you for your time,
Jakob Valtin

**Comment by stephaniestroka** [comment]
That makes sense, but when I run the build and, afterwards, run browserify on my host, I get 2 different sha256sum results. That's strange, right?
```
Step 12/12 : RUN sha256sum js/backup.js
 ---> Running in 940113a0bd6d
bf5b8e154472a60c89d559640222234cba574ee120dd98bcd14c854a914a5d9c  js/backup.js
 ---> ea814653eb9c
Removing intermediate container 940113a0bd6d
Successfully built ea814653eb9c
-bash-4.4$ browserify js/backup_in.js -o js/backup.js
-bash-4.4$ sha256sum js/backup.js
bc1eb4a784f2bf9dd463b8f9cba0f9546606a0bf75f973c837d880aa3d566161  js/backup.js
```

Is that a user error?

edit: a different browserify version might possibly produce a different output. My question is therefore more about whether it is important to verify that the SHA256sum stays the same across multiple docker runs or if you were aiming for a proof that the backup.js produced on docker is the same as the backup.js produced on a host.

**Comment by jvaltin** [comment]
> Is that a user error?

Almost certainly this is not a user error. It may not be deterministic or
it may not be reproducible across different build environments. Does it
result in a different sha256sum each build or only a different sha256sum
per (unique) build environment?

**Comment by stephaniestroka** [comment]
Ok, I just checked and on the one hand, different browserify versions produce different outputs, but, furthermore, even the same version has build-environment dependent things:
```
bash-4.4$ diff js/backup.js docker_build.out 
14878c14878
<   "_where": "/usr/local/lib/node_modules/browserify/node_modules/browserify-sign",
---
>   "_where": "/usr/lib/node_modules/browserify/node_modules/browserify-sign",
```

I guess that is simply used for additional debug information. Unfortunately that makes the sha256sum check less useful, but I'm going to merge your PR anyway, because it obviously doesn't hurt.

Thanks for your time and effort! :-)

**Pull request merged**